Client/Frontend/Home/PanelDataObservers.swift

```diff
@@ -48,28 +49,33 @@ class ActivityStreamDataObserver: DataObserver {
         events.forEach { NotificationCenter.default.removeObserver(self, name: $0, object: nil) }
     }
     
-    func invalidate(highlights: Bool) {
+    func refresh(highlights: Bool) {
+        guard !profile.isShutdown else {
+            return
+        }
         self.delegate?.willInvalidateDataSources()
 
-        let notify = {
+        // Highlights are cached for 15 mins
+        let invalidateHighlights = highlights ? true : (Date.now() - lastInvalidation > invalidationTime)
```

Comment:
Don't use `Date.now()` for timing; it can jump around, in which case your invalidation will be wrong. Grep the tree for `uptime`, and use offsets from uptime.